### PR TITLE
Attempted fix for landed state detection

### PR DESCRIPTION
**Pull request by JacksonElia**
Due to a couple of bugs, one of them being fixed in #58, we didn't actually log any data while the rocket was landed. You can see in the interest launch graph that the slope of the altitude line is still nonzero. This attempts to fix it, I will write tests for it tomorrow. 

You can see in the plot below that it doesn't actually detect landed state. This is to be expected as the launch data ends before the rocket landed.
image

I tried testing it with `winter_2023_Gyro_Comp.csv` but the mock sim almost immediately crashed without error (most likely in a different process), so that's some other problem we need to fix.

### CODE REVIEW

File: constants.py
Comment (harshil21) [inline constants.py:111]: I almost did this as well, but then our vertical velocity still showed -15m/s ish after landing (interest launch), so I was not sure if that's a good measure. Maybe if we give it enough time the velocity would come down, it's just that the data ended prematurely so I can't be sure.

**Review by harshil21 — COMMENTED** [review]
We should probably wait until #47 is merged and after testing with the physical IMU to check what velocity values we get (e.g. through a drop test)

File: constants.py
Comment (JacksonElia) [reply constants.py:111]: It showed this because we cut off the launch data before it actually landed I think. I think we should keep something like this and just guess the values, because worst case it doesn't do the state change and we just get all the data to log.

Comment (harshil21) [reply constants.py:111]: For the interest launch we definitely got touchdown (spike in accel). For purple nurple yeah idk it might have been cutoff

Comment (JacksonElia) [reply constants.py:111]: Maybe then for interest launch my file I was using for the sim got cut off at the end? You can see the slope of the altitude is not zero at the end.

Comment (harshil21) [reply constants.py:111]: I think you're using the wrong data? The graph in the readme shows LandedState.

Comment (harshil21) [reply constants.py:111]: oh it might have been will's new data also, I haven't looked at that

Comment (JacksonElia) [reply constants.py:111]: No my data is the same. the graph in the readme incorrectly detects the landed state, you can see it is still sloping down. Landed state should have a slope of 0.

Comment (harshil21) [reply constants.py:111]: > incorrectly detects

yeah it intentionally switches over at 15m, so the slope won't be zero. But we can confirm that we actually got touchdown because there is a spike in the acceleration values for both the launches in the landed state. (Run the sim from my branch with `-f` and `-l` and scroll through LandedState accel values)

Comment (harshil21) [reply constants.py:111]: But, yeah I get your point that we shouldn't switch over to LandedState arbitrarily like this. imo acceleration values should be the easiest (and safest?) way to detect landing, irrespective of height of the site. But maybe we can test with the IMU first to see what the rotated velocity values are like 1 second after landing

**Comment by harshil21** [comment]
I will change the csv file 😭 so the integration test passes

File: airbrakes/state.py
Comment (harshil21) [inline airbrakes/state.py:178]: maybe as a fallback we can have time in free fall state should not be greater than 80 seconds?

Comment (JacksonElia) [reply airbrakes/state.py:178]: I'm okay with that

File: airbrakes/state.py
Comment (harshil21) [inline airbrakes/state.py:183]: can we have this in combination with altitude < 15 just to be extra safe

Comment (JacksonElia) [reply airbrakes/state.py:184]: that kinda defeats the point of the fallback IMO

Comment (harshil21) [reply airbrakes/state.py:184]: ok. I was imagining a scenario where the chutes are deployed and the gust is so strong that it takes a really long time to land

Comment (JacksonElia) [reply airbrakes/state.py:184]: I talked to trent, he said the max amount of time he could ever see it being in the air (even if winds were insane) would be 180 seconds, so I'm just gonna make it that.

**Review by harshil21 — APPROVED** [review]

**Pull request merged**
